# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/OutputStreamManager.h
# Commit: 85ef72ebde43
# Full Hash: 85ef72ebde439c5de258af43faa9bd83a6e23b26
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-08-29 21:45:25
# Regressor Bug: 1573102
# File Overlap Count: 1
# Description:
#   Bug 1573102 - Remove mGraph from HTMLMediaElement::OutputMediaStream and OutputStreamManager. r=karlt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D43666
# ==============================================================================

diff -r f61dd73cac3a -r 85ef72ebde43 dom/media/mediasink/OutputStreamManager.h
--- a/dom/media/mediasink/OutputStreamManager.h	Thu Aug 29 13:32:31 2019 +0000
+++ b/dom/media/mediasink/OutputStreamManager.h	Thu Aug 29 13:32:50 2019 +0000
@@ -59,7 +59,7 @@
   NS_INLINE_DECL_THREADSAFE_REFCOUNTING(OutputStreamManager);
 
  public:
-  OutputStreamManager(MediaStreamGraphImpl* aGraph, nsIPrincipal* aPrincipal,
+  OutputStreamManager(MediaStreamGraph* aGraph, nsIPrincipal* aPrincipal,
                       AbstractThread* aAbstractMainThread);
   // Add the output stream to the collection.
   void Add(DOMMediaStream* aDOMStream);
@@ -153,7 +153,7 @@
   // Remove tracks sourced from aStream from all output streams.
   void RemoveTrack(SourceMediaStream* aStream);
 
-  const RefPtr<MediaStreamGraphImpl> mGraph;
+  const RefPtr<SourceMediaStream> mDummyStream;
   nsTArray<UniquePtr<OutputStreamData>> mStreams;
   nsTArray<UniquePtr<LiveTrack>> mLiveTracks;
   Canonical<PrincipalHandle> mPrincipalHandle;
